linode/manager
# upcoming: [DI-19818] - Added line graph to CloudPulse widgets for different metrics

## Description 📝
Line graph component is added in cloud pulse widget to showcase data for different metrics.


## Changes  🔄
List any change relevant to the reviewer.
1. CloudPulseLineGraph is added in the cloud pulse widget for metrics
2. LineGraph component is modified for full width legends as it was restricting to 85% width
3. Widget color palette is added for predefined colors in graph
4. Necessary type definitions are added to support metrics data
5. Added the metrics api call outside api-v4

## Target release date 🗓️
2nd August 2024

## Preview 📷
**Include a screenshot or screen recording of the change**

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
|![Screenshot 2024-07-25 at 3 17 07 PM](https://github.com/user-attachments/assets/8a2195b1-d29c-4ec1-9534-bbf632db40f2)|![Screenshot 2024-07-25 at 3 05 37 PM](https://github.com/user-attachments/assets/55f1b75c-0fc7-42fc-9c46-67dfc98e4c66)|
|![Screenshot 2024-07-25 at 3 05 44 PM](https://github.com/user-attachments/assets/1e7a4073-a5a1-410f-a6ad-68c213b0c89c)|![Screenshot 2024-07-25 at 3 05 56 PM](https://github.com/user-attachments/assets/f1acfd5e-b626-4975-8584-11ea180d918b)|
|![Screenshot 2024-07-25 at 7 30 44 PM](https://github.com/user-attachments/assets/dd429005-8450-4f5c-948d-a1a3c4918f26)|![Screenshot 2024-07-25 at 7 31 02 PM](https://github.com/user-attachments/assets/ad892dad-b042-4dad-86dd-9dde955be6c6)|




## How to test 🧪

1. Enable mock services as some endpoints are not ready to use.
2. Graph will be populated with dummy data so you can see the visualization & tool tips.

## As an Author I have considered 🤔

*Check all that apply*
